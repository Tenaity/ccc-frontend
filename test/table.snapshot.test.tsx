import { afterEach, expect, test } from "vitest"
import { cleanup, render } from "@testing-library/react"
import React from "react"

import {
  Table,
  TableBody,
  TableCaption,
  TableCell,
  TableFooter,
  TableHead,
  TableHeader,
  TableRow,
} from "@/components/ui/table"

afterEach(() => {
  cleanup()
})

test("basic table snapshot", () => {
  const { container } = render(
    <Table stickyHeader>
      <TableCaption>Team assignments</TableCaption>
      <TableHeader sticky>
        <TableRow>
          <TableHead>Nhân sự</TableHead>
          <TableHead className="text-right">Ca</TableHead>
        </TableRow>
      </TableHeader>
      <TableBody>
        <TableRow>
          <TableCell>Nguyễn Văn A</TableCell>
          <TableCell className="text-right">CA1</TableCell>
        </TableRow>
        <TableRow>
          <TableCell>Trần Thị B</TableCell>
          <TableCell className="text-right">CA2</TableCell>
        </TableRow>
      </TableBody>
      <TableFooter>
        <TableRow>
          <TableCell>Tổng</TableCell>
          <TableCell className="text-right">2 ca</TableCell>
        </TableRow>
      </TableFooter>
    </Table>
  )

  expect(container.firstChild).toMatchSnapshot()
})
