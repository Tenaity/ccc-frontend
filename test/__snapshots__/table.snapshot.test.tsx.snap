// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`basic table snapshot 1`] = `
<div
  class="relative w-full overflow-x-auto"
>
  <table
    class="w-full caption-bottom text-sm text-foreground"
    data-sticky-header="true"
  >
    <caption
      class="mt-4 text-sm text-muted-foreground"
    >
      Team assignments
    </caption>
    <thead
      class="[&_tr]:border-b sticky top-0 z-10 bg-background/95"
      data-sticky="true"
    >
      <tr
        class="border-b border-border/60 transition-colors hover:bg-muted/50"
      >
        <th
          class="h-11 px-3 text-left align-middle text-xs font-semibold text-muted-foreground"
        >
          Nhân sự
        </th>
        <th
          class="h-11 px-3 align-middle text-xs font-semibold text-muted-foreground text-right"
        >
          Ca
        </th>
      </tr>
    </thead>
    <tbody
      class="[&_tr:last-child]:border-b-0"
    >
      <tr
        class="border-b border-border/60 transition-colors hover:bg-muted/50"
      >
        <td
          class="p-3 align-middle text-sm"
        >
          Nguyễn Văn A
        </td>
        <td
          class="p-3 align-middle text-sm text-right"
        >
          CA1
        </td>
      </tr>
      <tr
        class="border-b border-border/60 transition-colors hover:bg-muted/50"
      >
        <td
          class="p-3 align-middle text-sm"
        >
          Trần Thị B
        </td>
        <td
          class="p-3 align-middle text-sm text-right"
        >
          CA2
        </td>
      </tr>
    </tbody>
    <tfoot
      class="bg-muted/60 font-medium text-foreground"
    >
      <tr
        class="border-b border-border/60 transition-colors hover:bg-muted/50"
      >
        <td
          class="p-3 align-middle text-sm"
        >
          Tổng
        </td>
        <td
          class="p-3 align-middle text-sm text-right"
        >
          2 ca
        </td>
      </tr>
    </tfoot>
  </table>
</div>
`;
